### DISCUSSION TITLE
How to automate circular rebalancing with RTL:

### DISCUSSION THREAD

**Discussion by BonzoDogDooDah** [Ride-The-Lightning/RTL#1325] [Q&A] [answered]
Hello,

We have tried the Torq, LNDG, and the LNDBOSS apps on our Umbrel Lightning Node in the hopes of implementing an automatic channel rebalancing system but so far they both do not wotk for us so we are still performing manual circular rebalancing using RTL. which works well for the most part and actually the only one that does for us. We would really love it if we could just use RTL to automate circular rebalancing but so far it does not appear to have this function, at least not yet? Please let us know if anyone here can help us with this?

Thank you. :)

Orange_Bonzo

**Comment by saubyk** [comment, marked as answer]
Hi @BonzoDogDooDah, circular rebalancing doesn't impact your node's overall channel balance. It enables you to just alter relative balances of the two channels involved in the operation. This may be helpful in an one-off need, to satisfy specific liquidity scenario, but it's hard to imagine a need to do such operations at scale.

Would you be able to shed some light on the use case you're trying to address by automating circular rebalancing. For example:
- What's the problem you typically address by circular rebalancing?
- How frequently do you need to do it?
- How do you make the decision on selecting the channels which you need to rebalance in this manner?
- How many channels does your node typically have?

**Comment by BonzoDogDooDah** [comment]
Hi Suheb

Thank you for your reply.

At the moment we only have 7 channels and 3 of them route sats into the node and 2 and sometimes 3 different channels will  then route sats out of the node. We find that our Acinq, Sunny Sarah, and Garlic channels require frequent refilling of their inbound liquidity which we will then do a circular rebalance using RTL and mostly use our LNBig channel and sometimes from one of our Nicehash channels to send the sats to either Acinq, Sarah, or Garlic. We used to use our C.otto.de channel in this way when it was working better but lately it does not do much of anything even after fee rate adjustments?

I see from your reply that our node appears to be lacking in proper management, as you say it should be a one-off need to do a circular rebalance and we do re balancing quite often otherwise our node does not route any sats at all?

Sunny Sarah takes all of the liquidity in a manner of minutes even after increasing our fee to 275 ppm?

I suppose you might say we are Nubie's in the woods in the land of lightning node management, Lol...

We do plan on increasing the amount of channels we have in the future.

Thank you for your support it is most appreciated :)

Orange_Bonzo


-- 
 Sent with Tuta; enjoy secure & ad-free emails: 
 https://tuta.com


Nov 13, 2023, 02:18 by ***@***.***:

**Reply by saubyk** [reply]
So ideally, if you aim to be a good routing node, your node channel selection should enable a bi-directional traffic.
If in your current situation, traffic only flows in one direction but not the other, which you keep compensating for by filling in liquidity from other channels, it's not a sustainable setup. As eventually, you will run out of liquidity on the channels, you're sourcing from.

I would recommend:
1. raising the fee rate to the liquidity sink to high enough level that it drains at a reasonable pace
2. source liquidity by getting more nodes to open channels with your node
3. close channels with nodes, which are not generating traffic to conserve your liquidity and deploy with more productive nodes
4. long term ensure that traffic is flowing in both directions such that you don't need to actively do liquidity management
5. ensure good uptime of your node by employing reasonably good hardware with power back etc. so that your node ratings improve

Hope this helps

**Reply by BonzoDogDooDah** [reply]
Hi Saubyk, Thank you for this very useful information. You said this: "your node channel selection should enable a bi-directional traffic." I assume you mean that by choosing the right peer to open a channel with it will become bi-directional thereby not requiring it to need rebalancing? I also assume that this is a big secret in the communtity as I have not found out how to do this yet. Would you be willing to shed some light on the way through to these greener pastures?  Your continued support is most appreciated,  Orange_Bonzo
